lib/pdk/tests/unit.rb

```diff
@@ -32,22 +32,13 @@ def self.invoke(report, options = {})
 
         result = command.execute!
 
-        # TODO: cleanup rspec and/or beaker output
-        # Iterate through possible JSON documents until we find one that is valid.
         json_result = nil
         json_result = [] if options.key?(:parallel)
 
-        result[:stdout].scan(%r{\{(?:[^{}]|(?:\g<0>))*\}}x) do |str|
-          begin
-            if options.key?(:parallel)
-              json_result.push(JSON.parse(str))
-            else
-              json_result = JSON.parse(str)
-              break
-            end
-          rescue JSON::ParserError
-            next
-          end
+        if options.key?(:parallel)
+          json_result.push(PDK::Util.find_valid_json_in(result[:stdout]))
```

Comment:
No problem @james-stocks, just pushed a new acceptance test. And it made me realize that it can only parallelize when there are multiple spec files. So I learned something about what I did 👍